Can't compile ESR with Clang18

The ESR version (115.x) of Firefox fails to compile with Clang18. Interestingly the latest version (124+) seems to compile just fine. Also, Clang17 compiles it without a hitch.

```
|  1:57.30     Building [===============>         ] 424/635: to_shmem, ash, neqo-commo...
|  1:57.30      Running `CARGO=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/recipe-sysroot-native/usr/bin/cargo CARGO_CRATE_NAME=gecko_profiler CARGO_MANIFEST_DIR=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/tools/profiler/rust-api CARGO_PKG_AUTHORS='The Mozilla Project Developers' CARGO_PKG_DESCRIPTION='' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MPL-2.0 CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=gecko-profiler CARGO_PKG_README=README.md CARGO_PKG_REPOSITORY='' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=0.1.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=1 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/release/deps:/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/recipe-sysroot-native/usr/lib' OUT_DIR=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/aarch64-poky-linux-gnu/release/build/gecko-profiler-0610fb31fa7a8372/out /home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/recipe-sysroot-native/usr/bin/rustc --crate-name gecko_profiler --edition=2018 tools/profiler/rust-api/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=2 -C panic=abort -C embed-bitcode=no --cfg 'feature="enabled"' -C metadata=e48180badcbb1c98 -C extra-filename=-e48180badcbb1c98 --out-dir /home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/aarch64-poky-linux-gnu/release/deps --target aarch64-poky-linux-gnu -C linker=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/build/cargo-linker -L dependency=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/aarch64-poky-linux-gnu/release/deps -L dependency=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/release/deps --extern bincode=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/aarch64-poky-linux-gnu/release/deps/libbincode-471141fe7b684c4b.rmeta --extern lazy_static=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/aarch64-poky-linux-gnu/release/deps/liblazy_static-6935d2c993f193c5.rmeta --extern mozbuild=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/aarch64-poky-linux-gnu/release/deps/libmozbuild-54ac2f0185eec587.rmeta --extern profiler_macros=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/release/deps/libprofiler_macros-3b9731097596afcc.so --extern serde=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/firefox-115.10.0/firefox-build-dir/aarch64-poky-linux-gnu/release/deps/libserde-3fa3c9153171b079.rmeta --cap-lints warn -L /home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/recipe-sysroot/usr/lib/rustlib/aarch64-poky-linux-gnu/lib --remap-path-prefix=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr=/usr/src/debug/firefox/115.10.0esr-r10 -Cpanic=abort -Clinker=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/wrapper/target-rust-ccld --sysroot=/home/gyuri/stuff/yocto/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/firefox/115.10.0esr/recipe-sysroot -Cembed-bitcode=yes -C codegen-units=1`
|  1:57.38 error[E0412]: cannot find type `SpliceableJSONWriter` in module `mozilla::baseprofiler`
|  1:57.38   --> tools/profiler/rust-api/src/gecko_bindings/glue.rs:18:46
|  1:57.38    |
|  1:57.38 18 |     json_writer: &mut mozilla::baseprofiler::SpliceableJSONWriter,
|  1:57.38    |                                              ^^^^^^^^^^^^^^^^^^^^ not found in `mozilla::baseprofiler`
|  1:57.38 error[E0412]: cannot find type `SpliceableJSONWriter` in module `mozilla::baseprofiler`
|  1:57.38   --> tools/profiler/rust-api/src/gecko_bindings/glue.rs:36:46
|  1:57.38    |
|  1:57.38 36 |     json_writer: &mut mozilla::baseprofiler::SpliceableJSONWriter,
|  1:57.38    |                                              ^^^^^^^^^^^^^^^^^^^^ not found in `mozilla::baseprofiler`
|  1:57.39 error[E0425]: cannot find function `gecko_profiler_marker_schema_stream` in module `bindings`
|  1:57.39   --> tools/profiler/rust-api/src/gecko_bindings/glue.rs:45:19
|  1:57.39    |
|  1:57.39 45 |         bindings::gecko_profiler_marker_schema_stream(
|  1:57.39    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.39 error[E0412]: cannot find type `SpliceableJSONWriter` in module `mozilla::baseprofiler`
|  1:57.39   --> tools/profiler/rust-api/src/json_writer.rs:13:58
|  1:57.39    |
|  1:57.39 13 | pub struct JSONWriter<'a>(&'a mut mozilla::baseprofiler::SpliceableJSONWriter);
|  1:57.39    |                                                          ^^^^^^^^^^^^^^^^^^^^ not found in `mozilla::baseprofiler`
|  1:57.39 error[E0412]: cannot find type `SpliceableJSONWriter` in module `mozilla::baseprofiler`
|  1:57.39   --> tools/profiler/rust-api/src/json_writer.rs:18:67
|  1:57.39    |
|  1:57.39 18 |     pub(crate) fn new(json_writer: &'a mut mozilla::baseprofiler::SpliceableJSONWriter) -> Self {
|  1:57.39    |                                                                   ^^^^^^^^^^^^^^^^^^^^ not found in `mozilla::baseprofiler`
|  1:57.39 error[E0425]: cannot find function `gecko_profiler_json_writer_int_property` in module `bindings`
|  1:57.39   --> tools/profiler/rust-api/src/json_writer.rs:26:23
|  1:57.39    |
|  1:57.39 26 |             bindings::gecko_profiler_json_writer_int_property(
|  1:57.39    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.39 error[E0425]: cannot find function `gecko_profiler_json_writer_float_property` in module `bindings`
|  1:57.39   --> tools/profiler/rust-api/src/json_writer.rs:39:23
|  1:57.39    |
|  1:57.39 39 |             bindings::gecko_profiler_json_writer_float_property(
|  1:57.39    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.39 error[E0425]: cannot find function `gecko_profiler_json_writer_bool_property` in module `bindings`
|  1:57.39   --> tools/profiler/rust-api/src/json_writer.rs:52:23
|  1:57.39    |
|  1:57.39 52 |             bindings::gecko_profiler_json_writer_bool_property(
|  1:57.39    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.39 error[E0425]: cannot find function `gecko_profiler_json_writer_string_property` in module `bindings`
|  1:57.39   --> tools/profiler/rust-api/src/json_writer.rs:65:23
|  1:57.39    |
|  1:57.39 65 |             bindings::gecko_profiler_json_writer_string_property(
|  1:57.39    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.39 error[E0425]: cannot find function `gecko_profiler_json_writer_null_property` in module `bindings`
|  1:57.39   --> tools/profiler/rust-api/src/json_writer.rs:79:23
|  1:57.39    |
|  1:57.39 79 |             bindings::gecko_profiler_json_writer_null_property(
|  1:57.39    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.39 error[E0412]: cannot find type `AutoProfilerLabel` in module `mozilla`
|  1:57.39   --> tools/profiler/rust-api/src/label.rs:19:51
|  1:57.39    |
|  1:57.39 19 | pub struct AutoProfilerLabel<'a>(&'a mut mozilla::AutoProfilerLabel);
|  1:57.39    |                                                   ^^^^^^^^^^^^^^^^^ not found in `mozilla`
|  1:57.39    |
|  1:57.39 help: consider importing this struct
|  1:57.39    |
|  1:57.39 11 + use crate::AutoProfilerLabel;
|  1:57.39    |
|  1:57.39 help: if you import `AutoProfilerLabel`, refer to it directly
|  1:57.39    |
|  1:57.39 19 - pub struct AutoProfilerLabel<'a>(&'a mut mozilla::AutoProfilerLabel);
|  1:57.39 19 + pub struct AutoProfilerLabel<'a>(&'a mut AutoProfilerLabel);
|  1:57.39    |
|  1:57.39 error[E0412]: cannot find type `AutoProfilerLabel` in module `mozilla`
|  1:57.39   --> tools/profiler/rust-api/src/label.rs:29:52
|  1:57.39    |
|  1:57.39 29 |         label: &mut std::mem::MaybeUninit<mozilla::AutoProfilerLabel>,
|  1:57.39    |                                                    ^^^^^^^^^^^^^^^^^ not found in `mozilla`
|  1:57.39    |
|  1:57.39 help: consider importing this struct
|  1:57.39    |
|  1:57.39 11 + use crate::AutoProfilerLabel;
|  1:57.39    |
|  1:57.40 help: if you import `AutoProfilerLabel`, refer to it directly
|  1:57.40    |
|  1:57.40 29 -         label: &mut std::mem::MaybeUninit<mozilla::AutoProfilerLabel>,
|  1:57.40 29 +         label: &mut std::mem::MaybeUninit<AutoProfilerLabel>,
|  1:57.40    |
|  1:57.40 error[E0425]: cannot find function `gecko_profiler_construct_label` in module `bindings`
|  1:57.40   --> tools/profiler/rust-api/src/label.rs:32:19
|  1:57.40    |
|  1:57.40 32 |         bindings::gecko_profiler_construct_label(
|  1:57.40    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.40 error[E0425]: cannot find function `gecko_profiler_destruct_label` in module `bindings`
|  1:57.40   --> tools/profiler/rust-api/src/label.rs:45:23
|  1:57.40    |
|  1:57.40 45 |             bindings::gecko_profiler_destruct_label(self.0);
|  1:57.40    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.40 error[E0412]: cannot find type `MarkerTiming` in module `mozilla`
|  1:57.40   --> tools/profiler/rust-api/src/marker/options.rs:23:57
|  1:57.40    |
|  1:57.40 23 | pub struct MarkerTiming(pub(crate) MaybeUninit<mozilla::MarkerTiming>);
|  1:57.40    |                                                         ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.40    |
|  1:57.40 help: consider importing this struct
|  1:57.40    |
|  1:57.40 8  + use crate::MarkerTiming;
|  1:57.40    |
|  1:57.40 help: if you import `MarkerTiming`, refer to it directly
|  1:57.40    |
|  1:57.40 23 - pub struct MarkerTiming(pub(crate) MaybeUninit<mozilla::MarkerTiming>);
|  1:57.40 23 + pub struct MarkerTiming(pub(crate) MaybeUninit<MarkerTiming>);
|  1:57.40    |
|  1:57.40 error[E0412]: cannot find type `MarkerTiming` in module `mozilla`
|  1:57.40   --> tools/profiler/rust-api/src/marker/options.rs:28:56
|  1:57.40    |
|  1:57.40 28 |         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.40    |                                                        ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.40    |
|  1:57.40 help: consider importing this struct
|  1:57.40    |
|  1:57.40 8  + use crate::MarkerTiming;
|  1:57.40    |
|  1:57.40 help: if you import `MarkerTiming`, refer to it directly
|  1:57.40    |
|  1:57.40 28 -         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.40 28 +         let mut marker_timing = MaybeUninit::<MarkerTiming>::uninit();
|  1:57.40    |
|  1:57.40 error[E0425]: cannot find function `gecko_profiler_construct_marker_timing_instant_at` in module `bindings`
|  1:57.40   --> tools/profiler/rust-api/src/marker/options.rs:30:23
|  1:57.40    |
|  1:57.40 30 |             bindings::gecko_profiler_construct_marker_timing_instant_at(
|  1:57.40    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.40 error[E0412]: cannot find type `MarkerTiming` in module `mozilla`
|  1:57.40   --> tools/profiler/rust-api/src/marker/options.rs:40:56
|  1:57.40    |
|  1:57.40 40 |         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.40    |                                                        ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.40    |
|  1:57.40 help: consider importing this struct
|  1:57.40    |
|  1:57.40 8  + use crate::MarkerTiming;
|  1:57.40    |
|  1:57.40 help: if you import `MarkerTiming`, refer to it directly
|  1:57.40    |
|  1:57.40 40 -         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.41 40 +         let mut marker_timing = MaybeUninit::<MarkerTiming>::uninit();
|  1:57.41    |
|  1:57.41 error[E0425]: cannot find function `gecko_profiler_construct_marker_timing_instant_now` in module `bindings`
|  1:57.41   --> tools/profiler/rust-api/src/marker/options.rs:42:23
|  1:57.41    |
|  1:57.41 42 |             bindings::gecko_profiler_construct_marker_timing_instant_now(
|  1:57.41    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.41 error[E0412]: cannot find type `MarkerTiming` in module `mozilla`
|  1:57.41   --> tools/profiler/rust-api/src/marker/options.rs:51:56
|  1:57.41    |
|  1:57.41 51 |         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.41    |                                                        ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.41    |
|  1:57.41 help: consider importing this struct
|  1:57.41    |
|  1:57.41 8  + use crate::MarkerTiming;
|  1:57.41    |
|  1:57.41 help: if you import `MarkerTiming`, refer to it directly
|  1:57.41    |
|  1:57.41 51 -         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.41 51 +         let mut marker_timing = MaybeUninit::<MarkerTiming>::uninit();
|  1:57.41    |
|  1:57.41 error[E0425]: cannot find function `gecko_profiler_construct_marker_timing_interval` in module `bindings`
|  1:57.41   --> tools/profiler/rust-api/src/marker/options.rs:53:23
|  1:57.41    |
|  1:57.41 53 |             bindings::gecko_profiler_construct_marker_timing_interval(
|  1:57.41    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.41 error[E0412]: cannot find type `MarkerTiming` in module `mozilla`
|  1:57.41   --> tools/profiler/rust-api/src/marker/options.rs:64:56
|  1:57.41    |
|  1:57.41 64 |         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.41    |                                                        ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.41    |
|  1:57.41 help: consider importing this struct
|  1:57.41    |
|  1:57.41 8  + use crate::MarkerTiming;
|  1:57.41    |
|  1:57.41 help: if you import `MarkerTiming`, refer to it directly
|  1:57.41    |
|  1:57.41 64 -         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.41 64 +         let mut marker_timing = MaybeUninit::<MarkerTiming>::uninit();
|  1:57.41    |
|  1:57.41 error[E0425]: cannot find function `gecko_profiler_construct_marker_timing_interval_until_now_from` in module `bindings`
|  1:57.41   --> tools/profiler/rust-api/src/marker/options.rs:66:23
|  1:57.41    |
|  1:57.41 66 |             bindings::gecko_profiler_construct_marker_timing_interval_until_now_from(
|  1:57.41    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.41 error[E0412]: cannot find type `MarkerTiming` in module `mozilla`
|  1:57.41   --> tools/profiler/rust-api/src/marker/options.rs:77:56
|  1:57.42    |
|  1:57.42 77 |         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.42    |                                                        ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.42    |
|  1:57.42 help: consider importing this struct
|  1:57.42    |
|  1:57.42 8  + use crate::MarkerTiming;
|  1:57.42    |
|  1:57.42 help: if you import `MarkerTiming`, refer to it directly
|  1:57.42    |
|  1:57.42 77 -         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.42 77 +         let mut marker_timing = MaybeUninit::<MarkerTiming>::uninit();
|  1:57.42    |
|  1:57.42 error[E0425]: cannot find function `gecko_profiler_construct_marker_timing_interval_start` in module `bindings`
|  1:57.42   --> tools/profiler/rust-api/src/marker/options.rs:79:23
|  1:57.42    |
|  1:57.42 79 |             bindings::gecko_profiler_construct_marker_timing_interval_start(
|  1:57.42    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.42 error[E0412]: cannot find type `MarkerTiming` in module `mozilla`
|  1:57.42   --> tools/profiler/rust-api/src/marker/options.rs:90:56
|  1:57.42    |
|  1:57.42 90 |         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.42    |                                                        ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.42    |
|  1:57.42 help: consider importing this struct
|  1:57.42    |
|  1:57.42 8  + use crate::MarkerTiming;
|  1:57.42    |
|  1:57.42 help: if you import `MarkerTiming`, refer to it directly
|  1:57.42    |
|  1:57.42 90 -         let mut marker_timing = MaybeUninit::<mozilla::MarkerTiming>::uninit();
|  1:57.42 90 +         let mut marker_timing = MaybeUninit::<MarkerTiming>::uninit();
|  1:57.42    |
|  1:57.42 error[E0425]: cannot find function `gecko_profiler_construct_marker_timing_interval_end` in module `bindings`
|  1:57.42   --> tools/profiler/rust-api/src/marker/options.rs:92:23
|  1:57.42    |
|  1:57.42 92 |             bindings::gecko_profiler_construct_marker_timing_interval_end(
|  1:57.42    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.42 error[E0425]: cannot find function `gecko_profiler_destruct_marker_timing` in module `bindings`
|  1:57.42    --> tools/profiler/rust-api/src/marker/options.rs:110:23
|  1:57.42     |
|  1:57.42 110 |             bindings::gecko_profiler_destruct_marker_timing(self.0.as_mut_ptr());
|  1:57.42     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.42 error[E0412]: cannot find type `StackCaptureOptions` in module `mozilla`
|  1:57.42    --> tools/profiler/rust-api/src/marker/options.rs:116:33
|  1:57.42     |
|  1:57.42 116 | pub type MarkerStack = mozilla::StackCaptureOptions;
|  1:57.42     |                                 ^^^^^^^^^^^^^^^^^^^ not found in `mozilla`
|  1:57.42 error[E0412]: cannot find type `MarkerSchema_Location` in module `mozilla`
|  1:57.42   --> tools/profiler/rust-api/src/marker/schema.rs:14:30
|  1:57.42    |
|  1:57.42 14 | pub type Location = mozilla::MarkerSchema_Location;
|  1:57.42    |                              ^^^^^^^^^^^^^^^^^^^^^ not found in `mozilla`
|  1:57.42 error[E0412]: cannot find type `MarkerSchema_Format` in module `mozilla`
|  1:57.42   --> tools/profiler/rust-api/src/marker/schema.rs:17:28
|  1:57.43    |
|  1:57.43 17 | pub type Format = mozilla::MarkerSchema_Format;
|  1:57.43    |                            ^^^^^^^^^^^^^^^^^^^ not found in `mozilla`
|  1:57.43 error[E0412]: cannot find type `MarkerSchema_Searchable` in module `mozilla`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:20:32
|  1:57.43    |
|  1:57.43 20 | pub type Searchable = mozilla::MarkerSchema_Searchable;
|  1:57.43    |                                ^^^^^^^^^^^^^^^^^^^^^^^ not found in `mozilla`
|  1:57.43 error[E0412]: cannot find type `MarkerSchema` in module `mozilla`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:30:50
|  1:57.43    |
|  1:57.43 30 |     pub(crate) pin: Pin<Box<MaybeUninit<mozilla::MarkerSchema>>>,
|  1:57.43    |                                                  ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.43    |
|  1:57.43 help: consider importing this struct
|  1:57.43    |
|  1:57.43 7  + use crate::MarkerSchema;
|  1:57.43    |
|  1:57.43 help: if you import `MarkerSchema`, refer to it directly
|  1:57.43    |
|  1:57.43 30 -     pub(crate) pin: Pin<Box<MaybeUninit<mozilla::MarkerSchema>>>,
|  1:57.43 30 +     pub(crate) pin: Pin<Box<MaybeUninit<MarkerSchema>>>,
|  1:57.43    |
|  1:57.43 error[E0412]: cannot find type `MarkerSchema` in module `mozilla`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:36:75
|  1:57.43    |
|  1:57.43 36 |         let mut marker_schema = Box::pin(std::mem::MaybeUninit::<mozilla::MarkerSchema>::uninit());
|  1:57.43    |                                                                           ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.43    |
|  1:57.43 help: consider importing this struct
|  1:57.43    |
|  1:57.43 7  + use crate::MarkerSchema;
|  1:57.43    |
|  1:57.43 help: if you import `MarkerSchema`, refer to it directly
|  1:57.43    |
|  1:57.43 36 -         let mut marker_schema = Box::pin(std::mem::MaybeUninit::<mozilla::MarkerSchema>::uninit());
|  1:57.43 36 +         let mut marker_schema = Box::pin(std::mem::MaybeUninit::<MarkerSchema>::uninit());
|  1:57.43    |
|  1:57.43 error[E0425]: cannot find function `gecko_profiler_construct_marker_schema` in module `bindings`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:39:23
|  1:57.43    |
|  1:57.43 39 |             bindings::gecko_profiler_construct_marker_schema(
|  1:57.43    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.43 error[E0412]: cannot find type `MarkerSchema` in module `mozilla`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:51:75
|  1:57.43    |
|  1:57.43 51 |         let mut marker_schema = Box::pin(std::mem::MaybeUninit::<mozilla::MarkerSchema>::uninit());
|  1:57.43    |                                                                           ^^^^^^^^^^^^ not found in `mozilla`
|  1:57.43    |
|  1:57.43 help: consider importing this struct
|  1:57.43    |
|  1:57.43 7  + use crate::MarkerSchema;
|  1:57.43    |
|  1:57.43 help: if you import `MarkerSchema`, refer to it directly
|  1:57.43    |
|  1:57.43 51 -         let mut marker_schema = Box::pin(std::mem::MaybeUninit::<mozilla::MarkerSchema>::uninit());
|  1:57.43 51 +         let mut marker_schema = Box::pin(std::mem::MaybeUninit::<MarkerSchema>::uninit());
|  1:57.43    |
|  1:57.43 error[E0425]: cannot find function `gecko_profiler_construct_marker_schema_with_special_front_end_location` in module `bindings`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:53:23
|  1:57.43    |
|  1:57.43 53 |             bindings::gecko_profiler_construct_marker_schema_with_special_front_end_location(
|  1:57.43    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.43 error[E0425]: cannot find function `gecko_profiler_marker_schema_set_chart_label` in module `bindings`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:66:23
|  1:57.43    |
|  1:57.43 66 |             bindings::gecko_profiler_marker_schema_set_chart_label(
|  1:57.43    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.43 error[E0425]: cannot find function `gecko_profiler_marker_schema_set_tooltip_label` in module `bindings`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:81:23
|  1:57.43    |
|  1:57.43 81 |             bindings::gecko_profiler_marker_schema_set_tooltip_label(
|  1:57.43    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.43     Building [===============>         ] 425/635: ash, neqo-common, selecto...
|  1:57.43 error[E0425]: cannot find function `gecko_profiler_marker_schema_set_table_label` in module `bindings`
|  1:57.43   --> tools/profiler/rust-api/src/marker/schema.rs:96:23
|  1:57.43    |
|  1:57.43 96 |             bindings::gecko_profiler_marker_schema_set_table_label(
|  1:57.43    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
|  1:57.43 error[E0425]: cannot find function `gecko_profiler_marker_schema_set_all_labels` in module `bindings`
|  1:57.43    --> tools/profiler/rust-api/src/marker/schema.rs:111:23
|  1:57.43     |
|  1:57.43 111 |             bindings::gecko_profiler_marker_schema_set_all_labels(
|  1:57.43     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `bindings`
```

--- a/meta-firefox/recipes-browser/firefox/firefox-esr/0004-fix-compilation-with-clang18.patch
+++ b/meta-firefox/recipes-browser/firefox/firefox-esr/0004-fix-compilation-with-clang18.patch
@@ -1,0 +1,72 @@
+Make sure that Firefox can be compiled with clang-18. Using the original
+bindgen crate, the compilation tripped on the following llvm commit:
+
+https://github.com/llvm/llvm-project/commit/7091ca1ae3a87479b6febdf1c3a324d707c633d9
+
+Upstream-Status: Inappropriate
+--- ./build/workspace-hack/Cargo.toml	2024-04-23 11:07:32.693320001 +0200
++++ ./build/workspace-hack/Cargo.toml	2024-04-23 11:07:45.550114940 +0200
+@@ -33,7 +33,7 @@
+ serde_derive = { features = ["default", "deserialize_in_place"], version = "1.0.66" }
+ quote = { features = ["default", "proc-macro"], version = "1.0" }
+ libc = { features = ["default", "std", "use_std"], version = "0.2" }
+-bindgen = { default-features = false, features = ["runtime"], version = "0.64" }
++bindgen = { default-features = false, features = ["runtime"], version = "0.69.4" }
+ 
+ [target."cfg(windows)".dependencies.winapi]
+ version = "0.3.6"
+--- ./netwerk/test/http3server/Cargo.toml	2024-04-23 11:11:53.996152320 +0200
++++ ./netwerk/test/http3server/Cargo.toml	2024-04-23 11:12:11.952865917 +0200
+@@ -27,7 +27,7 @@
+ 
+ # Make sure to use bindgen's runtime-loading of libclang, as it allows for a wider range of clang versions to be used
+ [build-dependencies]
+-bindgen = {version = "0.64", default-features = false, features = ["runtime"] }
++bindgen = {version = "0.69.4", default-features = false, features = ["runtime"] }
+ 
+ [[bin]]
+ name = "http3server"
+--- ./security/manager/ssl/builtins/Cargo.toml	2024-04-23 11:15:47.877422002 +0200
++++ ./security/manager/ssl/builtins/Cargo.toml	2024-04-23 11:15:56.513284263 +0200
+@@ -10,7 +10,7 @@
+ smallvec = { version = "1.9.0", features = ["const_new"] }
+ 
+ [build-dependencies]
+-bindgen = { default-features = false, features = ["runtime"], version = "0.64" }
++bindgen = { default-features = false, features = ["runtime"], version = "0.69.4" }
+ nom = "7.1.1"
+ 
+ [lib]
+--- ./tools/profiler/rust-api/Cargo.toml	2024-04-23 11:20:03.246319871 +0200
++++ ./tools/profiler/rust-api/Cargo.toml	2024-04-23 11:20:10.969195339 +0200
+@@ -14,7 +14,7 @@
+ 
+ [build-dependencies]
+ lazy_static = "1"
+-bindgen = {version = "0.64", default-features = false}
++bindgen = {version = "0.69.4", default-features = false}
+ mozbuild = "0.1"
+ 
+ [features]
+--- ./build/rust/bindgen/Cargo.toml	2024-04-23 11:29:11.516478991 +0200
++++ ./build/rust/bindgen/Cargo.toml	2024-04-23 11:29:19.829344946 +0200
+@@ -8,7 +8,7 @@
+ path = "lib.rs"
+ 
+ [dependencies.bindgen]
+-version = "0.64.0"
++version = "0.69.4"
+ default-features = false
+ 
+ [features]
+--- ./servo/components/style/Cargo.toml	2024-04-23 11:33:46.418046192 +0200
++++ ./servo/components/style/Cargo.toml	2024-04-23 11:34:00.513820782 +0200
+@@ -83,7 +83,7 @@
+ [build-dependencies]
+ lazy_static = "1"
+ log = { version = "0.4", features = ["std"] }
+-bindgen = {version = "0.64", optional = true, default-features = false}
++bindgen = {version = "0.69.4", optional = true, default-features = false}
+ regex = {version = "1.0", optional = true, default-features = false, features = ["perf", "std"]}
+ walkdir = "2.1.4"
+ toml = {version = "0.5", optional = true, default-features = false}

--- a/meta-firefox/recipes-browser/firefox/firefox_115.10.0esr.bb
+++ b/meta-firefox/recipes-browser/firefox/firefox_115.10.0esr.bb
@@ -10,7 +10,8 @@ SRC_URI += "git://github.com/hsivonen/packed_simd.git;protocol=https;branch=0_3_
             git://github.com/gfx-rs/d3d12-rs;protocol=https;branch=master;name=d3d12-rs;destsuffix=d3d12-rs \
             git://github.com/glandium/warp.git;protocol=https;branch=pemfile;name=warp;destsuffix=warp \
             git://github.com/gfx-rs/naga;protocol=https;branch=master;name=naga;destsuffix=naga \
-            file://debian-hacks/Work-around-bz-1775202-to-fix-FTBFS-on-ppc64el.patch"
+            file://debian-hacks/Work-around-bz-1775202-to-fix-FTBFS-on-ppc64el.patch \
+            file://0004-fix-compilation-with-clang18.patch"
 
 FILESEXTRAPATHS:prepend := "${THISDIR}/firefox-esr:"
 

--- a/meta-firefox/recipes-browser/firefox/firefox_crates_esr.inc
+++ b/meta-firefox/recipes-browser/firefox/firefox_crates_esr.inc
@@ -33,7 +33,7 @@ SRC_URI += "crate://crates.io/adler/1.0.2 \
             crate://crates.io/basic-toml/0.1.4 \
             crate://crates.io/bhttp/0.3.1 \
             crate://crates.io/bincode/1.3.3 \
-            crate://crates.io/bindgen/0.64.0 \
+            crate://crates.io/bindgen/0.69.4 \
             crate://crates.io/bit-set/0.5.3 \
             crate://crates.io/bit-vec/0.6.3 \
             crate://crates.io/bitflags/1.3.2 \
@@ -67,7 +67,7 @@ SRC_URI += "crate://crates.io/adler/1.0.2 \
             crate://crates.io/core-graphics-types/0.1.1 \
             crate://crates.io/core-graphics/0.22.3 \
             crate://crates.io/core-text/19.2.0 \
-            crate://crates.io/coreaudio-sys/0.2.12 \
+            crate://crates.io/coreaudio-sys/0.2.14 \
             crate://crates.io/coremidi-sys/3.1.0 \
             crate://crates.io/coremidi/0.6.0 \
             crate://crates.io/cose-c/0.1.5 \
@@ -296,7 +296,7 @@ SRC_URI += "crate://crates.io/adler/1.0.2 \
             crate://crates.io/pulse/0.3.0 \
             crate://crates.io/qlog/0.4.0 \
             crate://crates.io/quick-error/1.2.3 \
-            crate://crates.io/quote/1.0.23 \
+            crate://crates.io/quote/1.0.28 \
             crate://crates.io/rand_chacha/0.3.1 \
             crate://crates.io/rand_core/0.6.4 \
             crate://crates.io/rand/0.8.5 \
@@ -482,7 +482,7 @@ SRC_URI[base64-0.21.0.sha256sum] = "a4a4ddaa51a5bc52a6948f74c06d20aaaddb71924eab
 SRC_URI[basic-toml-0.1.4.sha256sum] = "7bfc506e7a2370ec239e1d072507b2a80c833083699d3c6fa176fbb4de8448c6"
 SRC_URI[bhttp-0.3.1.sha256sum] = "1300dca7a20730cce82c33fbf8795862556645ef5e9ee835390278c3fe1eb1d0"
 SRC_URI[bincode-1.3.3.sha256sum] = "b1f45e9417d87227c7a56d22e471c6206462cba514c7590c09aff4cf6d1ddcad"
-SRC_URI[bindgen-0.64.0.sha256sum] = "c4243e6031260db77ede97ad86c27e501d646a27ab57b59a574f725d98ab1fb4"
+SRC_URI[bindgen-0.69.4.sha256sum] = "a00dc851838a2120612785d195287475a3ac45514741da670b735818822129a0"
 SRC_URI[bit-set-0.5.3.sha256sum] = "0700ddab506f33b20a03b13996eccd309a48e5ff77d0d95926aa0210fb4e95f1"
 SRC_URI[bit-vec-0.6.3.sha256sum] = "349f9b6a179ed607305526ca489b34ad0a41aed5f7980fa90eb03160b69598fb"
 SRC_URI[bitflags-1.3.2.sha256sum] = "bef38d45163c2f1dde094a7dfd33ccf595c92905c8f8f4fdc18d06fb1037718a"
@@ -516,7 +516,7 @@ SRC_URI[core-foundation-sys-0.8.3.sha256sum] = "5827cebf4670468b8772dd191856768a
 SRC_URI[core-graphics-0.22.3.sha256sum] = "2581bbab3b8ffc6fcbd550bf46c355135d16e9ff2a6ea032ad6b9bf1d7efe4fb"
 SRC_URI[core-graphics-types-0.1.1.sha256sum] = "3a68b68b3446082644c91ac778bf50cd4104bfb002b5a6a7c44cca5a2c70788b"
 SRC_URI[core-text-19.2.0.sha256sum] = "99d74ada66e07c1cefa18f8abfba765b486f250de2e4a999e5727fc0dd4b4a25"
-SRC_URI[coreaudio-sys-0.2.12.sha256sum] = "f034b2258e6c4ade2f73bf87b21047567fb913ee9550837c2316d139b0262b24"
+SRC_URI[coreaudio-sys-0.2.14.sha256sum] = "f3120ebb80a9de008e638ad833d4127d50ea3d3a960ea23ea69bc66d9358a028"
 SRC_URI[coremidi-0.6.0.sha256sum] = "1a7847ca018a67204508b77cb9e6de670125075f7464fff5f673023378fa34f5"
 SRC_URI[coremidi-sys-3.1.0.sha256sum] = "79a6deed0c97b2d40abbab77e4c97f81d71e162600423382c277dd640019116c"
 SRC_URI[cose-0.1.4.sha256sum] = "72fa26cb151d3ae4b70f63d67d0fed57ce04220feafafbae7f503bef7aae590d"
@@ -745,7 +745,7 @@ SRC_URI[prost-derive-0.8.0.sha256sum] = "600d2f334aa05acb02a755e217ef1ab6dea4d51
 SRC_URI[pulse-0.3.0.sha256sum] = "88b7efc7519ba9fdddafe59e48ccdc6274a3fa6dbb4fc4a173dbc3f14860ab6d"
 SRC_URI[qlog-0.4.0.sha256sum] = "8777d5490145d6907198d48b3a907447689ce80e071b3d8a16a9d9fb3df02bc1"
 SRC_URI[quick-error-1.2.3.sha256sum] = "a1d01941d82fa2ab50be1e79e6714289dd7cde78eba4c074bc5a4374f650dfe0"
-SRC_URI[quote-1.0.23.sha256sum] = "8856d8364d252a14d474036ea1358d63c9e6965c8e5c1885c18f73d70bff9c7b"
+SRC_URI[quote-1.0.28.sha256sum] = "1b9ab9c7eadfd8df19006f1cf1a4aed13540ed5cbc047010ece5826e10825488"
 SRC_URI[rand_chacha-0.3.1.sha256sum] = "e6c10a63a0fa32252be49d21e7709d4d4baf8d231c2dbce1eaa8141b9b127d88"
 SRC_URI[rand_core-0.6.4.sha256sum] = "ec0be4795e2f6a28069bec0b5ff3e2ac9bafc99e6a9a7dc3547996c5c816922c"
 SRC_URI[rand-0.8.5.sha256sum] = "34af8d1a0e25924bc5b7c43c079c942339d8f0a8b57c39049bef581b46327404"

